.github/workflows/analysis.yml

```diff
@@ -38,6 +42,22 @@ jobs:
         run:  sudo apt-get install python3-setuptools $(./scripts/list-build-dependencies.sh -m apt -c gcc)
       - name: Install scan-build (Python version)
         run:  sudo pip3 install scan-build beautifulsoup4 html5lib
+
+      - name:  Prepare compiler cache
+        id:    prep-ccache
+        shell: bash
+        run:   |
+          mkdir -p "${CCACHE_DIR}"
+          echo "::set-output name=dir::$CCACHE_DIR"
+          echo "::set-output name=today::$(( $(date +%s) / 86400 ))"
+          echo "::set-output name=yesterday::$(( $(date +%s) / 86400 - 1 ))"
```

Comment:
I was looking at `date` options looking for alternatives and came up with some ideas like using `date +%y%j`, but your solutions will work better (no cache miss when new year rolls over).

I discovered there's an option of `date --date=yesterday +%Y%M%d`, but I don't know if it works cross-platform or only with new GNU - maybe worth checking out?

Reply:
Oh, nice find with the --yesterday flag! Will give it a try.